Support Linux 4.15?

While building vmm with linux-4.15, I encountered the following:
```
/tmp/vmm/vmx.c:515:8: error: unknown type name ‘vm_fault_t’
 static vm_fault_t vmm_vcpu_page_fault(struct vm_fault *vmf)
        ^~~~~~~~~~
scripts/Makefile.build:332: recipe for target '/tmp/vmm/vmx.o' failed
make[2]: *** [/tmp/vmm/vmx.o] Error 1
```

Would you consider to support linux-4.15, which is the default kernel used in Ubuntu Linux 18.04-LTS?

--- a/vmx.c
+++ b/vmx.c
@@ -4,6 +4,7 @@
 #include <linux/mm.h>
 #include <linux/anon_inodes.h>
 #include <linux/uaccess.h>
+#include <linux/version.h>
 
 #include "vmx.h"
 #include "config.h"
@@ -515,7 +516,11 @@ static long vmm_vcpu_ioctl(struct file *filp, unsigned int ioctl,
 	return r;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 17, 0)
+static int vmm_vcpu_page_fault(struct vm_fault *vmf)
+#else
 static vm_fault_t vmm_vcpu_page_fault(struct vm_fault *vmf)
+#endif
 {
 	struct vcpu *vcpu = vmf->vma->vm_file->private_data;
 	struct page *page;

